File: ios/ReactNativeSignView/RCTSignView.m
//
//  RCTSignView.m
//  ReactNativeSignView
//
//  Created by Harish on 20/02/19.
//  Copyright © 2019 Tekion. All rights reserved.
//

#import "UIKit/UIKit.h"
#import "RCTSignView.h"


@implementation RCTSignView 

@synthesize onSignAvailable;

- (void) commonInit{
    [super commonInit];
    super.delegate = self;
}

-(void)onSignatureChangesCb: (NSString*) base64DataOfSign{
    self.onSignAvailable(@{@"signature": base64DataOfSign});
}



@end

File: ios/ReactNativeSignView/RCTSignView.swift
//
//  RCTSignView.swift
//  ReactNativeSignView
//
//  Created by Harish on 17/02/19.
//  Copyright © 2019 Tekion. All rights reserved.
//

import Foundation

class RCTSignView: SignView, SignViewCallbacks{
    
    @objc var onSignAvailable: RCTDirectEventBlock?;
    
    override func commonInit() {
        super.commonInit();
        super.delegate = self;
    }
    
    func onSignatureChangesCb(_ base64DataOfSign: String) {
        onSignAvailable?(["signature":base64DataOfSign]);
    }
}
